src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -583,6 +588,44 @@ public override string GetText(MethodInfo method, object[] args)
             }
         }
 
+        private class ParamArgListFragment : NameFragment
+        {
+            private readonly int _maxStringLength;
+
+            public ParamArgListFragment(int maxStringLength)
+            {
+                _maxStringLength = maxStringLength;
+            }
+
+            public override string GetText(MethodInfo method, object[] args)
+            {
+                var sb = new StringBuilder();
+
+                if (args != null)
+                {
+                    sb.Append('(');
```

Comment:
Why does it make sense to skip adding `()` when args is null but not when args is empty?

Reply:
See ArgListFragment.
I changed the logic, which coincided with {a}. At the beginning, I always added brackets.

Reply:
There are two cases in which args will be null...

1. This is a simple test, specified by the `TestAttribute`, so no arguments are supplied at all.

2. A perverse use of `TestCaseSourceAttribute` returns null rather than an empty list.

The first case would seem to be the primary usage. If parentheses were added, the majority of test names in most applications would probably change, which would be bad.

Leaving aside the above "perverse" case, bear in mind that multiple sources, like `[TestCase]` and `[TestCaseSource]` maybe specified. Some of them might be empty. It's desirable that the names of the tests should be as similar as possible, given differences in arguments.

Reply:
Good, thanks!